src/RokuDeploy.ts

```diff
@@ -435,10 +435,12 @@ export class RokuDeploy {
      * @param port - the port that should be used for the request. defaults to 8060
      * @param timeout - request timeout duration in milliseconds. defaults to 150000
      */
-    public async pressHomeButton(host, port?: number, timeout?: number) {
+    public async pressHomeButton(host: string, port?: number, timeout?: number) {
         let options = this.getOptions();
         port = port ? port : options.remotePort;
         timeout = timeout ? timeout : options.timeout;
+        //convert hostname to ip address because ECP commands only work with IP addresses
+        host = await util.dnsLookup(host);
```

Comment:
Is there not a better spot for us to put this lookup than in this single call function?

Reply:
Well, this is the only place where it would actually fail. So while we _could_ translate it other places, we don't really need to, so is there a benefit?